django/db/migrations/autodetector.py

```diff
@@ -1694,6 +1700,24 @@ def _get_dependencies_for_generated_field(self, field):
                 )
         return dependencies
 
+    def _get_generated_field_dependencies_for_removed_field(
+        self, app_label, model_name, field_name
+    ):
+        dependencies = []
+        model_state = self.from_state.models[app_label, model_name]
+        generated_fields = (f for f in model_state.fields.values() if f.generated)
+        for field in generated_fields:
+            if field_name in models.Q(field.expression).referenced_base_fields:
```

Comment:
It seems like this pattern is already used in `_get_dependencies_for_generated_field` so not a huge deal but passing `field.expression: Expression` to a `Q` object (which is meant to be used for conditional expressions) so we can then call `referenced_base_fields` which then defers to `query.get_children_from_q` seems plain wrong given the created `Q` object is likely unusable (unless `field.expression` resolves to a boolean).

Looks like we should simply have an equivalent of `referenced_base_fields` that isn't a method of `Q`.